crates/dns-resolver/src/resolver.rs

```diff
@@ -181,14 +197,12 @@ impl TestResolver {
     }
 
     fn get<'a>(&'a self, full: &Name, record_type: RecordType) -> Result<Answer, DnsError> {
-        let mut authority = match full.is_fqdn() {
-            true => full.clone(),
-            false => {
-                let mut fqdn = full.clone();
-                fqdn.set_fqdn(true);
-                fqdn
-            }
+        let full_fqdn = {
+            let mut name = full.clone();
+            name.set_fqdn(true);
+            name
         };
+        let mut authority = full_fqdn.clone();
```

Comment:
Nit: it doesn't look like `authority` needs to clone `full_fqdn`? I'd also drop the nested scope for the `name` stuff if I were you, doesn't seem worth the syntactic overhead to me.